src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -158,17 +159,19 @@ private int getLowestEdgePriority(List<T> nodes) {
     return lowerPriority == null ? 0 : lowerPriority;
   }
 
-  private int getLowestOrder(List<T> nodes) {
+  private int getLowestOrder(List<T> nodes, int weight) {
```

Comment:
I'm confused by the naming and the semantics of this method. What is an order? Is it a weight? If yes, then call it `getLowestWeight()`, but then the passed parameter `weight` is confusing.

Can you add a comment explaining what this method does?